datadog_checks_base/datadog_checks/base/checks/openmetrics/v2/scraper.py

```diff
@@ -289,6 +305,11 @@ def parse_metrics(self):
 
             yield metric
 
+    @property
+    def parse_metric_families(self):
+        media_type = self._content_type.split(';')[0]
+        return self._parsers[media_type]
```

Comment:
Based on https://github.com/prometheus/prometheus/blob/v2.43.0/model/textparse/interface.go#L83-L90 couldn't we get rid of the parser mapping attribute and this could be simplified?

```suggestion
        return (
            parse_openmetrics
            if self.use_latest_spec or media_type == 'application/openmetrics-text'
            else parse_prometheus
        )
```